interfaces/cython/cantera/examples/onedim/flame_initial_guess.py

```diff
@@ -0,0 +1,160 @@
+"""
+A freely-propagating, premixed methane-air flame.
+Examples of saving and loading a flame and restarting
+with different initial guesses.
+
+Requires: cantera >= 3.0
+Keywords: combustion, 1D flow, flame speed, premixed flame, saving output
+"""
+import os
+import cantera as ct
+import pandas as pd
+import numpy as np
+
+data_directory = "flame_initial_guess_data"
+os.makedirs(data_directory, exist_ok=True)
+
+
+# Simulation parameters
+p = ct.one_atm  # pressure [Pa]
+Tin = 300.0  # unburned gas temperature [K]
+reactants = "CH4:0.45, O2:1.0, N2:3.76"
+
+width = 0.03  # m
+
+# Solution object used to compute mixture properties
+gas = ct.Solution("gri30.yaml")
+gas.TPX = Tin, p, reactants
+
+# Flame object
+f = ct.FreeFlame(gas, width=width)
+# These refine criteria are for a relatively quick convergence for the example
+# and may not be suitable for a publication-quality flame speed calculation
+refine_criteria = {"ratio": 3, "slope": 0.1, "curve": 0.2}
+f.set_refine_criteria(**refine_criteria)
+
+# Try to speed things up by loading the solution
+yaml_filepath = os.path.join(data_directory, "flame.yaml")
+try:
+    f.restore(yaml_filepath, name="solution", loglevel=0)
+except:
+    print("Couldn't load yaml solution, so starting from scratch")
+
+f.solve(loglevel=1, auto=True)
+print("mixture-averaged flame speed = {:7f} m/s\n".format(f.velocity[0]))
```

Comment:
New examples should use f-strings instead
```suggestion
print(f"mixture-averaged flame speed = {f.velocity[0]:7f} m/s\n")
```